src/client/viewSandbox.js

```diff
@@ -316,17 +325,22 @@ const init = () => {
           const selectedViewDescriptor = getSelectedViewDescriptor();
           if (selectedViewDescriptor && selectedViewDescriptor.schema) {
             return extensionView.getSettings().then(settings => {
-              const ajv = Ajv();
+              const ajv = Ajv({
+                loadSchema: loadSchema,
+                schemaId: 'auto'
+              });
               ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));
 
-              const matchesSchema = ajv.validate(selectedViewDescriptor.schema, settings);
+              ajv.compileAsync(selectedViewDescriptor.schema).then(validate => {
+                const matchesSchema = validate(settings);
 
-              if (matchesSchema) {
-                validateOutput.innerHTML = 'Valid';
-              } else {
-                validateOutput.innerHTML =
-                  '<span class="error">Settings object does not match schema</span>';
-              }
+                if (matchesSchema) {
+                  validateOutput.innerHTML = 'Valid';
+                } else {
+                  validateOutput.innerHTML =
+                    '<span class="error">Settings object does not match schema</span>';
+                }
+              });
```

Comment:
Should we add a catch here in case the async fails?

```
.catch(() => {
    validateOutput.innerHTML = '<span class="error">Cannot compile schema</span>';
});
```

Reply:
I think that's a good idea. I'll open a PR.